stdlib/public/RemoteInspection/TypeLowering.cpp

```diff
@@ -413,28 +412,42 @@ bool RecordTypeInfo::readExtraInhabitantIndex(remote::MemoryReader &reader,
 }
 
 BitMask RecordTypeInfo::getSpareBits(TypeConverter &TC, bool &hasAddrOnly) const {
+  // Start with all spare bits; we'll mask them out as we go...
   auto mask = BitMask::oneMask(getSize());
   switch (SubKind) {
   case RecordKind::Invalid:
-    return mask;    // FIXME: Should invalid have all spare bits?  Or none?  Does it matter?
+    // FIXME: Should invalid have all spare bits?  Or none?  Does it matter?
+    return mask;
   case RecordKind::Tuple:
   case RecordKind::Struct:
+    // Regular aggregates inherit spare bits from their fields
     break;
   case RecordKind::ThickFunction:
-    break;
+    // Thick functions have two fields:
+    // * Code pointer that might be signed and/or misaligned
+    // * Context that could be a tagged pointer
+    mask.makeZero();  // No spare bits
+    return mask;
   case RecordKind::OpaqueExistential: {
     // Existential storage isn't recorded as a field,
     // so we handle it specially here...
     int pointerSize = TC.targetPointerSize();
     BitMask submask = BitMask::zeroMask(pointerSize * 3);
     mask.andMask(submask, 0);
     hasAddrOnly = true;
+    // Mask the rest of the fields as usual...
     break;
   }
   case RecordKind::ClassExistential:
-    break;
-  case RecordKind::ExistentialMetatype:
-    break; // Field 0 is metadata pointer, a Builtin of type 'yyXf'
+    // Class existential is a data pointer that does expose spare bits
+    // ... so we can fall through ...
+  case RecordKind::ExistentialMetatype: {
+    // Initial metadata pointer has spare bits
+    auto mpePointerSpareBits = TC.getBuilder().getMultiPayloadEnumPointerMask();
+    mask.andMask(mpePointerSpareBits, 0);
+    mask.keepOnlyLeastSignificantBytes(8);
```

Comment:
Where does `8` come from? If it's meant to be the pointer size then this needs to use `TC.targetPointerSize()` instead.

Reply:
I believe `8` actually works correctly here, but for somewhat obscure reasons:  the pointer mask in this code is a uint64_t that only uses the lower 4 bytes to represent 32-bit masks, so in that case we know the upper 4 bytes are already zeroed.  But I agree that `TC.targetPointerSize()` would be clearer.

Not unrelated:  I'm sure there are plenty of little-endian assumptions in this code.  Fortunately, little-endian won and so no one cares about big-endian anymore.  🤞